### PR TITLE
Fix #4134 - Add support for iOS 14+ CarPlay Interface only

### LINKED ISSUE #4131: Deleted video still shows as added to playlist and shows open in playlist
### Description: 
Deleted video still shows as added to playlist and shows open in playlist

### Steps to Reproduce 
  1. Open two YT videos in individual tabs
  2. Play video on tab 1 and add it to playlist
  3. Switch to video on tab 2 and add it to playlist
  4. Open playlist and delete both videos that were added
  5. Close playlist and go back to tab2, shows Add to playlist button as expected
  6. Switch to tab 1, shows video is added to playlist
  7. Opening the button shows open in playlist. When playlist is opened its empty coz we deleted the video in step 4

**Actual result:** 
https://user-images.githubusercontent.com/17010094/131991061-4fce4545-7fd7-4c1c-8846-6649cbe4c87f.MP4

**Expected result:**
Update button for backgrounded tabs when playlist item is removed 

**Reproduces how often:** [Easily reproduced, Intermittent Issue]
Easy

**Brave Version:** 
1.31(21.9.1.16)

**Device details:** 
iPhone 7+ (iOS 14.5.1)

**Website problems only:**
- did you check with Brave Shields down? NA
- did you check in Safari/Firefox (WkWebView-based browsers)?  NA

### Additional Information
cc: @Brandon-T

### LINKED ISSUE #4134: Update Playlist for CarPlay iOS 14+
### Description: 
- CarPlay currently uses the iOS 13+ interface only. This gives use the least amount of customizability and a ton of bugs that Apple just won't fix.
- We should add support for iOS 14+ interfaces separately so if we ever drop iOS 13, we can just delete that section of the code. This allows us to have more customizability in terms of UI AND ability to detect whenever the car is plugged in/available.

### Steps to Reproduce 
  1. iOS 13 interface looks different from iOS 14. This is the only way to tell.

### PR DESCRIPTION
**Pull request by Brandon-T**
## Summary of Changes
- Fixes #4131, #4130, #4129, #4134
- Fixes adding duplicate items to car-play for iOS 13 and 14.
- Adds new interface support for iOS 14+.
- Adds support for UISceneDelegate which is required for CarPlay on iOS 14+.

This pull request fixes #4134

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

**Comment by Brandon-T** [comment]
I will localize it shortly, and move the first 3 fixes to a separate branch for `Release` as those aren't car-play related and will go out in release.

File: Client/Frontend/Browser/Playlist/Controllers/PlaylistCarplayControllerIOS14.swift [hunk 1]
```diff
@@ -0,0 +1,681 @@
...
+import CarPlay
+import MediaPlayer
+import Data
+import BraveShared
+import Shared
+
+private let log = Logger.browserLogger
+
+@available(iOS 14, *)
+class PlaylistCarplayControllerIOS14: NSObject {
```
Comment (iccub) [inline Client/Frontend/Browser/Playlist/Controllers/PlaylistCarplayControllerIOS14.swift:17]: SonarCloud gonna hate you for it :D
it's ok, we drop iOS13 soon and can address it, maybe create a followup ticket to not forget about it?

File: Client/Frontend/Browser/Playlist/Managers & Cache/CPTemplateApplicationSceneDelegate.swift [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    func sceneDidDisconnect(_ scene: UIScene) {
+        if scene.session.configuration.name == "CPTemplateSceneConfiguration" {
+            log.debug("Template application scene did disconnect.")
+        }
+    }
+    
+    func sceneDidBecomeActive(_ scene: UIScene) {
+        if scene.session.configuration.name == "CPTemplateSceneConfiguration" {
```
Comment (iccub) [inline Client/Frontend/Browser/Playlist/Managers & Cache/CPTemplateApplicationSceneDelegate.swift]: I would extract `CPTemplateSceneConfiguration` to a constant

**Review by iccub — COMMENTED** [review]
looks good overall

**Review by iccub — APPROVED** [review]
Retested on simulator, works for me, good job :)

**Review by iccub — APPROVED** [review]

**Pull request merged**